### PR TITLE
Pack everything in an instantiable struct instead of using global vars.

### PR DESCRIPTION
**Pull request by hectorj**
Hi!

Love what you've done, the "prepared statements" are a nice idea.

This PR removes most global vars (just kept the Pool, as I don't see a reason to not keep it that way), and allow for several instances in one application (for different output writers for example).

Tell me what you think

### CODE REVIEW

File: nanolog_test.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func TestSetWriter(t *testing.T) {
 	SetWriter(buf)
 
 	// simulate some logging
-	defaultLogWriter.w.WriteByte(35)
+	defaultLogWriter.(*logWriter).w.WriteByte(35)
```
Comment (ScottMansfield) [inline nanolog_test.go]: These are all very ugly and don't really add anything to the tests. They should probably just be instantiating a new LogWriter each time.

File: nanolog_test.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ func TestSetWriter(t *testing.T) {
 	SetWriter(buf)
 
 	// simulate some logging
-	w.WriteByte(35)
+	defaultLogWriter.(*logWriter).w.WriteByte(35)
```
Comment (ScottMansfield) [inline nanolog_test.go]: these don't really add much to the tests. Each one should probably be creating its own LogWriter instance.

**Review by ScottMansfield — COMMENTED** [review]
Overall, great change. I'm not super clear on how people would manage their log lines with their own LogWriter, but that would be up to the user I suppose.

File: nanolog_test.go (hunk 1 above)
Comment (hectorj) [reply nanolog_test.go]: I wanted to keep your tests on the default interface, but agreed, will do.

Comment (hectorj) [reply nanolog_test.go]: https://github.com/ScottMansfield/nanolog/pull/2/commits/ac5d0da7d2338fc391c38c146dd5fd099905954b

**Comment by ScottMansfield** [comment]
awesome, looks good. Thanks, Hector!

**Pull request merged**